### PR TITLE
Ignore tables without table_type parameters while loading all iceberg table from Glue and Hive catalog (#1331)

### LINKED ISSUE #1150: Error: `table_type` missing from table parameters when loading table from Hive metastore
### Apache Iceberg version

main (development)

### Please describe the bug 🐞

I'm (a user of [tap-iceberg](https://github.com/shaped-ai/tap-iceberg) is) running into the following error when trying to load a Hive table using `pyiceberg`.

```
pyiceberg.exceptions.NoSuchPropertyException: Property table_type missing, could not determine type: bronze.my_iceberg_table
```

The call in question is https://github.com/shaped-ai/tap-iceberg/blob/38064b3aaca5394ba1482970e790d3e2f6020946/tap_iceberg/tap.py#L94.

It seems the loaded table is missing the table type parameter in

https://github.com/apache/iceberg-python/blob/d587e6724685744918ecf192724437182ad01abf/pyiceberg/catalog/hive.py#L329-L331

?

Thanks in advance if this turns out to be user error 😃

### LINKED ISSUE #1331: Error: table_type missing from table parameters while only loading iceberg table from Hive metaStore
### Apache Iceberg version

main (development)

### Please describe the bug 🐞

New problem related to #1145, if there are tables with no "table_type" in table.parameters, cli could not list tables properly but runs into an exception. Or should every hive table have "table_type" default value in table.parameters?

https://github.com/apache/iceberg-python/blob/b2f0a9e5cd7dd548e19cdcdd7f9205f03454369a/pyiceberg/catalog/hive.py#L647-L655

### PR DESCRIPTION
**Pull request by gitzwz**
Fixes #1331
Fixes #1150

### CODE REVIEW

**Comment by gitzwz** [comment]
same problem with #1258

File: pyiceberg/catalog/hive.py [hunk 1]
```diff
@@ -651,7 +651,7 @@ def list_tables(self, namespace: Union[str, Identifier]) -> List[Identifier]:
                 for table in open_client.get_table_objects_by_name(
                     dbname=database_name, tbl_names=open_client.get_all_tables(db_name=database_name)
                 )
-                if table.parameters[TABLE_TYPE].lower() == ICEBERG
+                if table.parameters.get("table_type", "").lower() == ICEBERG
```
Comment (HonahX) [inline pyiceberg/catalog/hive.py:654]:
```suggestion
                if table.parameters.get(TABLE_TYPE, "").lower() == ICEBERG
```
Just one minor thing: How about using constant `TABLE_TYPE` defined in `catalog/__init__.py` here? I just realized I forgot to mention this in #1258

Comment (gitzwz) [reply pyiceberg/catalog/hive.py]: > Just one minor thing: How about using constant `TABLE_TYPE` defined in `catalog/__init__.py` here? I just realized I forgot to mention this in #1258

@HonahX Thanks for reviewing this! I have fixed this.

**Review by HonahX — APPROVED** [review]

**Review by Fokko — APPROVED** [review]
Thanks for fixing this @gitzwz, and thanks for the review @HonahX 🙌

**Pull request merged**